pkg/api/types.go

```diff
@@ -584,6 +584,8 @@ type DownwardAPIVolumeFile struct {
 	Path string `json:"path"`
 	// Required: Selects a field of the pod: only annotations, labels, name and  namespace are supported.
 	FieldRef ObjectFieldSelector `json:"fieldRef"`
+	// Required: Selects a field of the pod; only name and namespace are supported.
+	ConfigDataRef *ConfigDataSelector `json:"configDataRef"`
```

Comment:
This follows the same behavior as FieldRef in environment - FieldRef should be more flexible.  It does mean we *have* to allow field ref selection that allows map fields to be loaded, which means we need to sort through how that happens.  It's probably straight up string hardcoding for now.